lib/middleware/rescued_subdomain_elevator.rb

```diff
@@ -0,0 +1,22 @@
+require "apartment/elevators/subdomain"
+
+module Apartment
+  module Elevators
+    class RescuedSubdomain < Subdomain
+      def call(env)
+        super
+      rescue ::Apartment::SchemaNotFound
+        [302, {"Location" => root_url(env)}, []]
+      end
+
+      private
+
+      def root_url(env)
+        host = env["HTTP_HOST"]
+        scheme = env["rack.url_scheme"]
+        root_url = host.gsub("#{subdomain(host)}.", "")
+        [scheme, root_url].join("://")
+      end
```

Comment:
This is the most generic way I could come up with to handle this in Rack, but we could also just throw something in ENV["ROOT_HOST"] and redirect there..